models/sale_order.py

```diff
@@ -18,8 +19,14 @@ def _compute_delivery_status(self):
             if not order.picking_ids or all(p.state == 'cancel' for p in order.picking_ids):
                 order.delivery_status = False
             elif all(p.state in ['done', 'cancel'] for p in order.picking_ids):
-                order.delivery_status = 'full'
+                if all(float_compare(line.qty_delivered, line.product_uom_qty, precision_rounding=line.product_uom.rounding) == 0 for line in order.order_line):
```

Comment:
Hello @shortcut1337 Ici j'ai l'impression qua sur les SO, je ne passe pas par cette condition, alors que pour les PO, ça marche bien. 
Ici je veux dire, si les qty_livrées et qty_commandées sont ==, alors la livraisons est full...

Reply:
> BL/BR cancelled

C'est quoi BL / BR ?

Note pour moi même (je viens d'écrire ça pour m'y retrouver, du coup au lieu de le jeter..):
Conceptuellement, le code existant faisait:
- Si j'ai pas de picking ou si y'en a mais qu'ils sont tous cancel, je n'ai pas de delivery status
- Sinon si j'ai des pickings et qu'ils sont tous soit en DONE ou soit en CANCEL, je suis fully delivered
- Sinon, si j'ai au moins un picking en DONE (mais que donc ils sont pas tous en done ou cancel), je suis partial delivered
- Sinon (j'ai donc un picking au moins, et aucun n'est en DONE et ils ne sont pas tous cancel) je suis en pending delivery

Reply:
Ca semble fonctionner chez moi, je passe bien par là en jouant avec les delivery order, tu as un step to reproduce qui ne fonctionne pas ?

Reply:
CAS 1:
1. Create a SO with 2 products.
2. Deliver one product.
3. Create a backorder
4. Cancel the backorder
5. Edit the SO of the product that you're not gonna deliver.
6. Status is "Partially Delivered"


CAS 2 = Cas 1 où mais je switch les étapes 4 et 5 :
1. Create a SO with 2 products.
2. Deliver one product.
3. Create a backorder
4. Edit the SO of the product that you're not gonna deliver.
5. Cancel the backorder
6. Status is "Fully delivered"

Il me semble que pour les PO, je peux le faire dans les 2 sens...

Reply:
Je vais déjà merger pour push ce soir, comme ça c'est fait.
J'expliquerai de faire comme dans le CAS 2, et pas le 1, et d'un point de vue Business ça s'explique (c'est au vendeur de décider ce qu'on livre ou non, avec accord du client)
en attendant qu'on trouve l'origine du problème.